homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,179 @@
+        return [HVAC_MODE_AUTO]
+
+    def set_hvac_mode(self, hvac_mode: str):
+        """Set new target hvac mode."""
+        return
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation if supported."""
+        return None
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return float(self._data.get("T_TARGET", 0))
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if target_temperature is None:
+            return False
+
+        try:
+            await self._base_update_handler.async_set_target_temperature(
+                self._data["ID"], target_temperature
+            )
+            self._data["T_TARGET"] = target_temperature
+            return True
+        except Exception as update_err:  # pylint: disable=broad-except
+            _LOGGER.error("Setting target temperature failed: %s", update_err)
+            return False
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode."""
+        # HEATAREA_MODE: 0=Auto, 1=Tag, 2=Nacht
+        if self._data["HEATAREA_MODE"] == 1:
+            return PRESET_COMFORT
+        if self._data["HEATAREA_MODE"] == 2:
+            return PRESET_ECO
+        return PRESET_NONE
+
+    @property
+    def preset_modes(self):
+        """Return available preset modes."""
+        return [PRESET_NONE, PRESET_COMFORT, PRESET_ECO]
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new operation mode."""
+        # HEATAREA_MODE: 0=Auto, 1=Tag, 2=Nacht
+        heatarea_mode = 0
+        if preset_mode == PRESET_COMFORT:
+            heatarea_mode = 1
+        elif preset_mode == PRESET_ECO:
+            heatarea_mode = 2
+
+        try:
+            await self._base_update_handler.async_set_heatarea_mode(
+                self._data["ID"], heatarea_mode
+            )
+            self._data["HEATAREA_MODE"] = heatarea_mode
+            if heatarea_mode == 1:
+                self._data["T_TARGET"] = self._data["T_HEAT_DAY"]
+            elif heatarea_mode == 2:
+                self._data["T_TARGET"] = self._data["T_HEAT_NIGHT"]
+            return True
+        except Exception as update_err:  # pylint: disable=broad-except
+            _LOGGER.error("Setting target temperature failed: %s", update_err)
+            return False
+
+    @property
+    def device_state_attributes(self):
+        """Return the optional state attributes."""
+        data = super().device_state_attributes or {}
+        data["ISLOCKED"] = self._data.get("ISLOCKED", False)
+        data["LOCK_AVAILABLE"] = self._data.get("LOCK_AVAILABLE", False)
+        data["BLOCK_HC"] = self._data.get("BLOCK_HC", False)
+        return data
+
+    def update(self):
+        """Get latest data from Alpha2 base."""
+        self._base_update_handler.update()
```

Comment:
do you want each entity to be able to call for an update, if the goal of the central updater is to make sure the api doesn't get overloaded then you don't!

Reply:
Remove update method.